### PR TITLE
Document database migrations

### LINKED ISSUE #390: Document how to create database migrations
It's currently undocumented, and I'm having a hard time remembering how this works :)

### PR DESCRIPTION
**Pull request by 0livd**
Closes #390

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -39,14 +39,26 @@ Accessing dev server
...
+--------------------------
+
+In case you need to modify the database schema, first update the models in ihatemoney/models.py.
+Then run the following command to create a new database revision file::
+
+  make create-database-revision
+
+If your changes are simple enough, the generated script will be populated with
+the necessary migrations steps.
+For complex migrations, these steps need to be written manually. Existing revision files
```
Comment (JocelynDelalande) [inline docs/contributing.rst]: Actually, this sentence about complex migrations is a bit vague, and IMHO reveals a slight lack in our tooling : a simple command to create empty migration. Current command to do that is :

```
python -m ihatemoney.manage db revision -d ihatemoney/migrations
```

(given that we are already in virtualenv, which is not an assumption we make for other dev-related commands).

Maybe it is good time to add a `make create-empty-database-revision` and adapt doc, WDYT ?

Comment (almet) [reply docs/contributing.rst]: +1 for the Makefile command.

Comment (JocelynDelalande) [reply docs/contributing.rst]: @0livd gentle bump :-). any time for this ?

Comment (0livd) [reply docs/contributing.rst]: Yep, I've just added the new makefile command :)

File: docs/contributing.rst [hunk 2]
```diff
@@ -39,14 +39,28 @@ Accessing dev server
...
+Create database migrations
+--------------------------
+
+In case you need to modify the database schema, first update the models in ihatemoney/models.py.
+Then run the following command to create a new database revision file::
+
+  make create-database-revision
+
+If your changes are simple enough, the generated script will be populated with
+the necessary migrations steps.
```
Comment (JocelynDelalande) [inline docs/contributing.rst]: Suggestion of adding « You can edit the generated script. eg: to add data migrations. »

File: CHANGELOG.rst [hunk 3]
```diff
@@ -35,6 +35,7 @@ Added
 - Improve settings documentation styling (#251)
 - Add a ihatemoney delete-project command to delete a project (#375)
 - Add nice 404 error pages (#379)
+- Document database migrations (#390)
```
Comment (JocelynDelalande) [inline CHANGELOG.rst]: Please rebase and make sure it is in the new version section of the changelog :-)

**Comment by almet** [comment]
hey, let us know when you've rebased so we can merge!

**Comment by almet** [comment]
Gasp. I did some changes to my local branch and then merged the wrong branch :x

**Comment by almet** [comment]
This was ugly, sorry. I've lost some git-fu these last months I believe.

**Pull request merged**